src/components/snapshots/takeSnapshot.ts

```diff
@@ -24,89 +32,128 @@ import {
 import { getDirectoryFromPath } from './useSnapshot'
 import DBConnect from '../../../src/components/databaseConnect'
 import config from '../../config'
 import { DateTime } from 'luxon'
 const asyncRimRaf = promisify(rimraf)
 import { createDefaultDataFolders } from '../files/createDefaultFolders'
+import { getArchiveFolders } from '../files/helpers'
+
+const TEMP_SNAPSHOT_FOLDER_NAME = '__tempSnapshot'
+const TEMP_ARCHIVE_FOLDER_NAME = '__tempArchive'
 
 const takeSnapshot: SnapshotOperation = async ({
   snapshotName = DEFAULT_SNAPSHOT_NAME,
   optionsName = DEFAULT_OPTIONS_NAME,
   options: inOptions,
   extraOptions = {},
+  isArchiveSnapshot = false,
 }) => {
   // Ensure relevant folders exist
   createDefaultDataFolders()
 
-  try {
-    console.log(`taking snapshot, name: ${snapshotName}`)
+  let archiveInfo: ArchiveInfo = null
 
-    const options = await getOptions(optionsName, inOptions, extraOptions)
+  const options = await getOptions(optionsName, inOptions, extraOptions)
 
-    const newSnapshotFolder = path.join(SNAPSHOT_FOLDER, snapshotName)
-    // Remove and create snapshot folder
-    await asyncRimRaf(newSnapshotFolder)
-    execSync(`mkdir ${newSnapshotFolder}`)
+  try {
+    console.log(`Taking ${isArchiveSnapshot ? 'Archive ' : ''}snapshot, name: ${snapshotName}`)
 
-    // Write snapshot/database to folder
-    if (options.usePgDump) {
-      // The quick way, using pg_dump -- whole database only
-      console.log('Dumping database...')
-      execSync(
-        `pg_dump -U postgres tmf_app_manager --format=custom -f ${newSnapshotFolder}/database.dump`
-      )
-      // This plain-text .sql script is NOT used for re-import, but could be
-      // useful for debugging when dealing with troublesome snapshots
-      execSync(
-        `pg_dump -U postgres tmf_app_manager --format=plain --inserts --clean --if-exists -f ${newSnapshotFolder}/database.sql`
-      )
-      console.log('Dumping database...done')
+    const tempFolder = path.join(SNAPSHOT_FOLDER, TEMP_SNAPSHOT_FOLDER_NAME)
+    const tempArchiveFolder = path.join(SNAPSHOT_FOLDER, TEMP_ARCHIVE_FOLDER_NAME)
+    await fse.emptyDir(tempFolder)
 
-      // Copy ALL files
-      console.log('Exporting files...')
-      execSync(`cp -a '${FILES_FOLDER}'/. '${newSnapshotFolder}/files'`)
-      console.log('Exporting files...done')
+    // Write snapshot/database to folder
+    if (!isArchiveSnapshot) {
+      if (options.usePgDump) {
+        // The quick way, using pg_dump -- whole database only
+        console.log('Dumping database...')
+        execSync(
+          `pg_dump -U postgres tmf_app_manager --format=custom -f ${tempFolder}/database.dump`
+        )
+        // This plain-text .sql script is NOT used for re-import, but could be
+        // useful for debugging when dealing with troublesome snapshots
+        // execSync(
+        //   `pg_dump -U postgres tmf_app_manager --format=plain --inserts --clean --if-exists -f ${tempFolder}/database.sql`
```

Comment:
Commented out this `.sql` script as we don't use it for anything and it's actually a really large file wasting a lot of space. We can un-comment this if we ever need to debug the database dumps.